carma_wm/src/WorldModelUtils.cpp

```diff
@@ -168,6 +168,24 @@ lanelet::ConstLaneletOrAreas getAffectedLaneletOrAreas(const lanelet::Points3d&
       std::unordered_set<lanelet::Lanelet> filtered = filterSuccessorLanelets(possible_lanelets, affected_lanelets, lanelet_map, routing_graph);
       ROS_DEBUG_STREAM("Got successor lanelets of size: " << filtered.size());
       affected_lanelets.insert(filtered.begin(), filtered.end());
+
+      if (affected_lanelets.empty() && !possible_lanelets.empty())
+      {
+        ROS_DEBUG_STREAM("Checking if it is the edge case where only last point falls on a valid (correct direction) lanelet");
+        for (auto llt: possible_lanelets)
+        {
+          ROS_DEBUG_STREAM("Evaluating lanelet: " << llt.id());
+          lanelet::BasicLineString2d gf_dir_line({gf_pts[idx - 1].basicPoint2d(), gf_pts[idx].basicPoint2d()});
```

Comment:
We are trying to get ingress/egress. This logic would only run after processing the last point of the offset points, and only if it was not able to detect any lanelets by then, as a last resort. I think it will not run on interior lanelets if offset points are given correctly.